### PR TITLE
//regen creates vanilla decorations without error

### LINKED ISSUE #1615: Errors in console on regen. Result was fine though
### Server Implementation

Paper

### Server Version

1.18.1

### Describe the bug

Marked a bit area that has been severely damaged and run `regen`. Took about 80 seconds to complete and the result seemed just fine. However I saw a few errors popped up in console. Not sure if a big deal or not. Probably not.

### To Reproduce

**Could not reproduce (yet).**

- Play with hugh spheres of stone and tnt and let it destroy the environment (yes, it takes quite a while for thousands of TNT to detonate).
- Mark area of about 500x500 with expand vert
- Run `//regen`
- wait

### Expected behaviour

Regenerate without throwing errors I guess.

### Screenshots / Videos

Did not took any since I noticed as things were already done, sorry.

### Error log (if applicable)

https://paste.gg/p/anonymous/3d7748ba25c84aa3b4e03f7fe8b84f67

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/59d492c9c7f647ebbf9efa4f018b4362

### Fawe Version

2.0.2-SNAPSHOT-94;cf0a21d

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

I hope I did not miss any relevant information. If so please let me know and I come back to it asap.

Cheers

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
Fix: Feature placement on regen should create chunk
Fixes #1615

For older versions: Same as https://github.com/IntellectualSites/FastAsyncWorldEdit-Adapters/pull/58

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [ ] I tested my changes and approved their functionality.
- [x] I ensured my changes do not break other parts of the code
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by NotMyFault** [comment]
Are there reliable steps to reproduce the issue? Otherwise I'd need to approve it provisionally.

**Comment by PierreSchwang** [comment]
> Are there reliable steps to reproduce the issue? Otherwise I'd need to approve it provisionally.

Same steps as with your 1.18.2 adapter bug, e.g. when regenerating a witch hut etc

**Comment by NotMyFault** [comment]
> > Are there reliable steps to reproduce the issue? Otherwise I'd need to approve it provisionally.
> 
> Same steps as with your 1.18.2 adapter bug, e.g. when regenerating a witch hut etc
> 
> Edit: Otherwise take a look at the MC source, when MC passes true it expects the chunk to be created if it does not exist yet.

Hm, I can regenerate a swamp hut fine on 118

**Comment by PierreSchwang** [comment]
Example from MC source which throws the named error in the issue
grafik

Mainly relevant for StructureFeatureManager, which creates structures like Witch Huts etc

File: worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/regen/PaperweightRegen.java [hunk 1]
```diff
@@ -312,8 +312,12 @@ public Biome getUncachedNoiseBiome(int biomeX, int biomeY, int biomeZ) {
         ) {
             // redirect to LevelChunks created in #createChunks
             @Override
-            public ChunkAccess getChunk(int x, int z, ChunkStatus chunkstatus, boolean flag) {
-                return getChunkAt(x, z);
+            public ChunkAccess getChunk(int x, int z, ChunkStatus chunkstatus, boolean create) {
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/regen/PaperweightRegen.java:315]: Isn't the ChunkStatus nullable?

**Comment by NotMyFault** [comment]
Conflicts

**Comment by PierreSchwang** [comment]
Will address after the ticking_list PR, as that changes the adapter (jar) as well

**Comment by NotMyFault** [comment]
@PierreSchwang Do you want to pick this PR up anytime soon? Otherwise I'd head forward to it tomorrow or the day after.

**Review by NotMyFault — APPROVED** [review]
Thanks for your contribution. I'll carry over @SirYwell review if the build passes.

**Comment by PierreSchwang** [comment]
> Can you rebase the adapters jar again to not bundle 1.15.2 anymore? See [IntellectualSites/FastAsyncWorldEdit-Adapters#61](https://github.com/IntellectualSites/FastAsyncWorldEdit-Adapters/pull/61) for the reason.

That should be covered in 9773db541348883a005050e3fb0b630ef44baf7b - at least i built the adapters and replaced them

**Comment by NotMyFault** [comment]
> > Can you rebase the adapters jar again to not bundle 1.15.2 anymore? See [IntellectualSites/FastAsyncWorldEdit-Adapters#61](https://github.com/IntellectualSites/FastAsyncWorldEdit-Adapters/pull/61) for the reason.
> 
> That should be covered in [9773db5](https://github.com/IntellectualSites/FastAsyncWorldEdit/commit/9773db541348883a005050e3fb0b630ef44baf7b) - at least i built the adapters and replaced them

The jar in this PR does ship 1.15.2
.

**Pull request merged**